app/api/routes/image_queries.py

```diff
@@ -152,50 +168,28 @@ async def post_image_query(
-                confidence_threshold=confidence_threshold,
+                confidence=new_image_query.result.confidence, confidence_threshold=confidence_threshold
             ):
                 logger.debug("Motion detection confidence is high enough to return.")
-                app_state.db_manager.create_iqe_record(record=new_image_query)
+                app_state.db_manager.create_iqe_record(iq=new_image_query)
                 return new_image_query
 
-    image_query = None
+    # -- Edge-model Inference --
     if not require_human_review and edge_inference_manager.inference_is_available(detector_id=detector_id):
-        detector_metadata: Detector = get_detector_metadata(detector_id=detector_id, gl=gl)
         logger.debug(f"Local inference is available for {detector_id=}. Running inference...")
         results = edge_inference_manager.run_inference(detector_id=detector_id, image=image)
         confidence = results["confidence"]
 
-        if edge_only or _is_confident_enough(
-            confidence=confidence,
-            detector_metadata=get_detector_metadata(detector_id=detector_id, gl=gl),
-            confidence_threshold=confidence_threshold,
-        ):
+        if edge_only or _is_confident_enough(confidence=confidence, confidence_threshold=confidence_threshold):
             if edge_only:
-                logger.info(
-                    "Edge-only mode is enabled on this detector. The edge model's answer will be returned "
-                    "regardless of confidence. {detector_id=}"
+                logger.debug(
+                    f"Edge-only mode enabled - will not escalate to cloud, regardless of confidence. {detector_id=}"
                 )
             else:
-                logger.info(f"Edge detector confidence is high enough to return. {detector_id=}")
-
-            if patience_time is None:
-                patience_time = constants.DEFAULT_PATIENCE_TIME  # Default patience time
-
-            if confidence_threshold is None:
-                confidence_threshold = detector_metadata.confidence_threshold  # Use detector's confidence threshold
-
-            mode = detector_metadata.mode
-            if mode == ModeEnum.BINARY:
-                result_type = ResultTypeEnum.binary_classification
-                results["label"] = "NO" if results["label"] else "YES"  # Map false / 0 to "YES" and true / 1 to "NO"
-            elif mode == ModeEnum.COUNT:
-                result_type = ResultTypeEnum.counting
-            elif mode == ModeEnum.MULTI_CLASS:
-                result_type = ResultTypeEnum.multi_classification
-            else:
-                raise ValueError(f"Got unrecognized detector mode: {mode}")
+                logger.debug(f"Edge detector confidence is high enough to return. {detector_id=}")
+
+            result_type, results = _mode_to_result_type(detector_metadata.mode, results)
+            patience_time = patience_time or constants.DEFAULT_PATIENCE_TIME
```

Comment:
This might create a small bug - this will go to the False case if `patience_time` is 0.0, right?

Reply:
good point! thanks!